invenio/templates/configurations/logstash.yaml

```diff
@@ -52,67 +52,133 @@ data:
+            target => "data"
+            # Download static database and point to it
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
+      grok {
+         match => { "url" => ["^/(api/)?records?/%{DATA}/(media-)?files/"]}
+         add_field => {"[data][request_category]" => "files" }
+      }
+      # add a field with the category "legacy", based on the url
+      grok {
+         match => { "url" => ["^/communities/about/%{DATA}/", "^/collection/user-%{INT}", "^/communities/%{DATA}/about", "^/communities/%{DATA}/edit", "^/communities/%{DATA}/curate", "^/communities/%{DATA}/search", "^/collection/%{DATA}", "^/deposit/%{DATA} ", "^/record/%{DATA}", "^/record/%{DATA}/export/%{DATA}", "^/records/%{DATA}/export/%{DATA}", "^/record/%{DATA}/files/%{DATA}", "^/record/%{DATA}/preview/%{DATA}", "^/deposit$", "^/deposit/new$", "^/record/%{DATA}/thumb%{DATA}", "^/record/%{DATA}/formats%{DATA}", "^/dev$", "^/faq$", "^/features$", "^/whatsnew$", "^/about$", "^/contact$", "^/policies$", "^/privacy-policy$", "^/terms$", "^/donate$"]}
```

Comment:
The match needs a bit of refining. There is a `"^/record/%{DATA}"`, that matches quite a lot of other patterns. I'm not sure if that is intended

Reply:
Thanks for the comment and the help, updated the PR 👍